build/gulp/tasks/test-projects.tsx

```diff
@@ -9,23 +9,12 @@ import config from '../../../config'
 import * as tmp from 'tmp'
 
 const { paths } = config
-
-const log = msg => {
-  console.log()
-  console.log('='.repeat(80))
-  console.log('CRA TS:', msg)
-  console.log('='.repeat(80))
-}
+let packageFilename
```

Comment:
don't like this broken encapsulation between the tasks, especially given that all can be used in isolation. Would prefer correctness there at the first place

Reply:
I'm not so familiar with Gulp, do you have a better approach there?

Reply:
Nevermind, `yarn pack` is quite fast so I moved it back to each step.

Reply:
@layershifter, thanks - unfortunately, there is built-in support in gulp to produce result from task (just in a form of artefacts, but this is definitely not what your question about:) As alternative, we might consider cache semantics, but absolutely agree that it is better to split, especially if this is a quick step 👍